src/Symfony/Component/Config/Resource/ReflectionClassResource.php

```diff
@@ -151,12 +151,66 @@ private function generateSignature(\ReflectionClass $class)
             }
         } else {
             foreach ($class->getMethods(\ReflectionMethod::IS_PUBLIC | \ReflectionMethod::IS_PROTECTED) as $m) {
-                yield preg_replace('/^  @@.*/m', '', $m);
-
                 $defaults = [];
+                $parametersWithUndefinedConstants = [];
                 foreach ($m->getParameters() as $p) {
-                    $defaults[$p->name] = $p->isDefaultValueAvailable() ? $p->getDefaultValue() : null;
+                    if (!$p->isDefaultValueAvailable()) {
+                        $defaults[$p->name] = null;
+
+                        continue;
+                    }
+
+                    if (!$p->isDefaultValueConstant() || \defined($p->getDefaultValueConstantName())) {
+                        $defaults[$p->name] = $p->getDefaultValue();
+
+                        continue;
+                    }
+
+                    $defaults[$p->name] = $p->getDefaultValueConstantName();
+                    $parametersWithUndefinedConstants[$p->name] = true;
                 }
+
+                if (!$parametersWithUndefinedConstants) {
+                    yield preg_replace('/^  @@.*/m', '', $m);
+                } else {
+                    yield $m->getDocComment();
+                    yield $m->getName();
+                    yield (int) $m->isAbstract();
+                    yield (int) $m->isFinal();
+                    yield (int) $m->isStatic();
+                    yield (int) $m->isPublic();
+                    yield (int) $m->isPrivate();
+                    yield (int) $m->isProtected();
+                    yield (int) $m->returnsReference();
+
+                    foreach ($m->getParameters() as $p) {
+                        if (!isset($parametersWithUndefinedConstants[$p->name])) {
+                            yield (string) $p;
+                        } else {
+                            yield (int) $p->isOptional();
+                            yield (int) ($hasType = \PHP_VERSION_ID >= 70000 && $p->hasType());
+
+                            if ($hasType) {
+                                yield (string) $p->getType();
```

Comment:
casting to string is only for 7.0: on 7.1+, we should use $p->getType()->getName() instead as the former is deprecated

Reply:
be careful than not all types are instanced of `ReflectionNamedType` (see `ReflectionUnionType`)

Reply:
you're right
that's something we will fix when adding support for PHP8 - not only this place will be affected.